genetic_algorithm/basic_string.py

```diff
@@ -54,19 +64,27 @@ def select(
     population_score: list[tuple[str, float]],
     genes: list[str],
 ) -> list[str]:
-    """Select the second parent and generate new population"""
+    """
+    Select the second parent and generate new population.
+    >>> random.seed(42)
+    >>> select(("123456", 8), [("abcdef", 4), ("ghijkl", 5)], list("ABCDEF"))
+    ['ghijkl', '1234B6']
+    """
     pop = []
     # Generate more children proportionally to the fitness score.
     child_n = int(parent_1[1] * 100) + 1
     child_n = 10 if child_n >= 10 else child_n
     for _ in range(child_n):
-        parent_2 = population_score[random.randint(0, N_SELECTED)][0]
-
-        child_1, child_2 = crossover(parent_1[0], parent_2)
-        # Append new string to the population list.
-        pop.append(mutate(child_1, genes))
-        pop.append(mutate(child_2, genes))
-    return pop
+        # Randomly select the second parent from the population.
```

Comment:
Please revert.  Please either add tests or change the algorithm but not both in a single pull request.